src/webots/nodes/WbLidar.cpp

```diff
@@ -351,33 +351,70 @@ void WbLidar::copyAllLayersToSharedMemory() {
   }
 }
 
+/* To be removed */
+#include <sys/timeb.h>
+uint64_t Time_uclock(void) {
+  struct timespec tv;
+
+  if (clock_gettime(CLOCK_REALTIME, &tv) != 0)
+    return 0;
+
+  return tv.tv_sec  * 1000000 + (tv.tv_nsec / 1000.0);
+}
+
 void WbLidar::updatePointCloud(int minWidth, int maxWidth) {
+  uint64_t t0_chrono = Time_uclock(); /* To be removed */
   WbLidarPoint *lidarPoints = pointArray();
   const float *image = lidarImage();
-
   const int resolution = actualHorizontalResolution();
+  const int numberOfLayer = actualNumberOfLayers();
   const double w = width();
-  const double time = WbSimulationState::instance()->time() / 1000.0;
 
-  for (int i = 0; i < actualNumberOfLayers(); ++i) {
-    double phi = 0;
-    if (actualNumberOfLayers() > 1)  // to avoid division by zero
-      phi = verticalFieldOfView() / 2 - i * (verticalFieldOfView() / (actualNumberOfLayers() - 1));
-    const double sinPhi = sin(phi + mCurrentTiltAngle);
-    const double cosPhi = cos(phi + mCurrentTiltAngle);
-    for (int j = minWidth; j < maxWidth; ++j) {
-      double theta = actualFieldOfView() / 2 - j * (actualFieldOfView() / (w - 1));
-      if (mIsActuallyRotating)
-        theta = -((double)j / (double)resolution) * 2 * M_PI;
-      const int index = resolution * i + j;
+  const double dt = -((double)mRefreshRate / 1000.0) / w;
+  const double t0 = WbSimulationState::instance()->time() / 1000.0 + minWidth * dt;
+
+  const double dphi = (numberOfLayer > 1) ? (-verticalFieldOfView() / (numberOfLayer - 1)) : 0.0;
+  const double cosdPhi = cos(dphi), sindPhi = sin(dphi);
```

Comment:
We usually do not put multiple statements in one line.
```suggestion
  const double cosdPhi = cos(dphi);
  const double sindPhi = sin(dphi);
```